IPIP/0379-delegated-ipns-http-api.md

```diff
@@ -0,0 +1,82 @@
+# IPIP-379: Delegated IPNS HTTP API
+
+- Start Date: 2023-02-13
+- Related Issues:
+  - https://github.com/ipfs/specs/issues/343
+  - https://github.com/ipfs/specs/pull/337
+  - https://github.com/ipfs/specs/pull/377
+
+## Summary
+
+This IPIP specifies a set of HTTP APIs to offload naming system onto another process or server.
+
+## Motivation
+
+Expanding on the motivations of [IPIP-337 HTTP Delegated Routing](0337-delegated-routing-http-api.md), the work here concentrates on delegation of _naming system_ over HTTP APIs. Naming is part of the core IPFS DHT functionality.
+The performance of naming system over the IPFS DHT can suffer from long delays due to churn of records and quorum requirements.
+
+## HTTP API Specification
+
+See [Delegated IPNS HTTP API](../routing/DELEGATED_IPNS_HTTP.md) specification.
+
+## Design rationale
+
+The rationale for delegated IPNS over HTTP APIs closely follows the reasoning listed in [IPIP-337](0337-delegated-routing-http-api.md#design-rationale).
+
+The document proposes the following:
+- Use of HTTP semantics for publication and resolution of naming records.
+- Preference for human-readable request and response encoding, such as JSON format
+- Optional backward compatibility support for the existing ProtocolBuffer format using `Content-Type: application/vnd.ipfs.ipns-record`
+- Use of extra headers in `OPTIONS` response to communicate the supported capabilities and limitations, namely:
+  - `X-Ipns-Allow-Max-Size` -- to signal maximum supported IPNS record size
+  - `X-Ipns-Allow-Protobuf` -- to signal whether the server supports ProtocolBuffer formatted records.
+- Streaming interaction is not supported.
+
+### User benefit
+
+The ability of offload naming onto another process or server via an idiomatic and simple HTTP API opens up an opportunity to not only reduce the burden of naming system management, but also introduce alternative naming systems across other routing systems, namely IPNI.
+
+### Compatibility
```

Comment:
TODO(before we ratify this):
- [ ] clients use `application/vnd.ipfs.ipns-record` ([IANA](https://github.com/ipfs/in-web-browsers/issues/209), [multicodec](https://github.com/multiformats/multicodec/pull/312)) no matter what router type is used for GET/PUT of IPNS record
- [ ] clients can use trustless gateway and  /routing/v1 endpoints  interchangeably, and a single service can share IPNS record cache across all supported routers DHT/Pubsub/HTTP